pkg/cloudprovider/cloud.go

```diff
@@ -47,6 +47,9 @@ type Interface interface {
 	ScrubDNS(nameservers, searches []string) (nsOut, srchOut []string)
 	// HasClusterID returns true if a ClusterID is required and set
 	HasClusterID() bool
+	// KeyManagementService provides a named key management service supported by the cloud. Returns an error
+	// if the named provider was not found.
+	KeyManagementService(name string) (KeyManagementService, error)
```

Comment:
Is there a maximum of one supported KMS per cloud by design or should those who support more than one hide that  inside of the Encrypt()/Decypt() calls?